app/javascript/react/screens/App/common/NotificationList/NotificationList.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  TimedToastNotification,
+  ToastNotificationList
+} from 'patternfly-react';
+
+class NotificationList extends React.Component {
+  render() {
+    const { notifications, removeNotificationAction } = this.props;
+
+    return (
+      <ToastNotificationList>
+        {notifications &&
+          notifications.map((notification, index) => (
+            <React.Fragment>
+              <TimedToastNotification
+                key={index}
+                type={notification.notificationType}
+                persistent={notification.persistent}
+                timerdelay={notification.timerdelay}
+                onDismiss={removeNotificationAction({ key: index })}
+                onMouseLeave={removeNotificationAction({ key: index })}
```

Comment:
since you're actually calling these two functions above (removeNotificationAction & removeNotificationAction), you need something like

```
onDismiss={() => removeNotificationAction(...)}
```

otherwise, those actions will fire immediately after rendering